### PR TITLE
add missing unique index on suse tables

### PR DESCRIPTION
**Pull request by rjmateus**
Signed-off-by: Ricardo Mateus <rmateus@suse.com>

## What does this PR change?

During the development of "ISS Revamp: Data dumper implementation #12968" we found some tables with missing unique key constraints.
This PR will add a unique index to the following tables:
- suseproductchannel
- susemdkeyword
- suseupgradepath

## GUI diff

No difference.

## Documentation
- No documentation needed, schema changes.

## Test coverage
- No tests. Adding missing unique index.

## Links

Fixes https://github.com/SUSE/spacewalk/issues/13517
Tracks:
- Port to Manager 4.1 https://github.com/SUSE/spacewalk/pull/13591

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests" 
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests" (Test skipped, there are no changes to test)
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: schema/spacewalk/common/tables/suseMdKeyword.sql [hunk 1]
```diff
@@ -24,3 +24,6 @@ suseMdKeyword
 ;
 
 CREATE SEQUENCE suse_mdkeyword_id_seq;
+
+CREATE UNIQUE INDEX susemdkeyword_lable_uq
```
Comment (mcalmer) [inline schema/spacewalk/common/tables/suseMdKeyword.sql:28]: typo in lable => label

File: schema/spacewalk/upgrade/susemanager-schema-4.2.4-to-susemanager-schema-4.2.5/802-add-missing-uq-index.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+CREATE UNIQUE INDEX IF NOT EXISTS suseproductchannel_product_id_channel_id_uq
+ON suseProductChannel (product_id, channel_id);
+
+CREATE UNIQUE INDEX IF NOT EXISTS susemdkeyword_lable_uq
```
Comment (mcalmer) [inline schema/spacewalk/upgrade/susemanager-schema-4.2.4-to-susemanager-schema-4.2.5/802-add-missing-uq-index.sql:5]: typo as above

**Review by mcalmer — CHANGES_REQUESTED** [review]
small typo

File: schema/spacewalk/common/tables/suseMdKeyword.sql (hunk 1 above)
Comment (rjmateus) [reply schema/spacewalk/common/tables/suseMdKeyword.sql]: thanks, MC. It's now corrected

File: schema/spacewalk/upgrade/susemanager-schema-4.2.4-to-susemanager-schema-4.2.5/802-add-missing-uq-index.sql (hunk 2 above)
Comment (rjmateus) [reply schema/spacewalk/upgrade/susemanager-schema-4.2.4-to-susemanager-schema-4.2.5/802-add-missing-uq-index.sql]: thanks, MC. It's now corrected

**Review by admd — APPROVED** [review]
:+1:

**Comment by admd** [comment]
Just noticed some tests are failing so please take a look.

**Review by mcalmer — APPROVED** [review]

**Review by mcalmer — CHANGES_REQUESTED** [review]
Sorry, had to revert my approval. It seems that java tests are failing. Please check why. It seems to be related to the change

**Comment by rjmateus** [comment]
Test was corrected. They were inserting duplicated data on table SUSEProductChannel

**Review by mcalmer — APPROVED** [review]

**Pull request merged**